examples/latency/index-async-await.js

```diff
@@ -0,0 +1,149 @@
+function newClient(clientStub, apiKey) {
+    c = new dgraph.DgraphClient(clientStub);
+    if (apiKey != "") {
+        c.setSlashApiKey(apiKey);
+    }
+    return c;
+}
+
+// Drop All - discard all data and start from a clean slate.
+async function dropAll(dgraphClient) {
+    await dgraphClient.alter({ dropAll: true });
+}
+
+// Set schema.
+async function setSchema(dgraphClient) {
+    const schema = `
+        name: string @index(exact) .
+        age: int .
+        married: bool .
+        loc: geo .
+        dob: datetime .
+    `;
+    await dgraphClient.alter({ schema: schema });
+}
+
+// Create data using JSON.
+async function createData(dgraphClient) {
+    // Create a new transaction.
+    const txn = dgraphClient.newTxn();
+    try {
+        // Create data.
+        const p = {
+            uid: "_:blank-0",
+            name: "Alice",
+            age: 26,
+            married: true,
+            loc: {
+                type: "Point",
+                coordinates: [1.1, 2],
+            },
+            dob: new Date(1980, 1, 1, 23, 0, 0, 0),
+            friend: [
+                {
+                    name: "Bob",
+                    age: 24,
+                },
+                {
+                    name: "Charlie",
+                    age: 29,
+                }
+            ],
+            school: [
+                {
+                    name: "Crown Public School",
+                }
+            ]
+        };
+
+        // Run mutation.
+        const assigned = await txn.mutate({ setJson: p });
+
+        // Commit transaction.
+        await txn.commit();
+
+        // Get uid of the outermost object (person named "Alice").
+        // Assigned#getUidsMap() returns a map from blank node names to uids.
+        // For a json mutation, blank node names "blank-0", "blank-1", ... are used
+        // for all the created nodes.
+        console.log(`Created person named "Alice" with uid = ${assigned.data.uids["blank-0"]}\n`);
+
+        console.log("All created nodes (map from blank node names to uids):");
+        Object.keys(assigned.data.uids).forEach((key) => console.log(`${key} => ${assigned.data.uids[key]}`));
+        console.log();
+    } finally {
+        // Clean up. Calling this after txn.commit() is a no-op
+        // and hence safe.
+        await txn.discard();
+    }
+}
+
+// Query for data.
+async function queryData(dgraphClient) {
+    // Run query.
+    const query = `query all($a: string) {
+        all(func: eq(name, $a)) {
+            uid
+            name
+            age
+            married
+            loc
+            dob
+            friend {
+                name
+                age
+            }
+            school {
+                name
+            }
+        }
+    }`;
+    const vars = { $a: "Alice" };
+
+    console.time('Query client latency');
+    const res = await dgraphClient.newTxn().queryWithVars(query, vars);
+    console.timeEnd('Query client latency');
```

Comment:
Suggestion: we can consider printing the query on console here along with the latency.

Reply:
Good point. I've added a log for the query (it collapses the query to a single line):

<pre>
<strong>Query: query all($a: string) { all(func: eq(name, $a)) { uid name age married loc dob friend { name age } school { name } } } Vars: { '$a': 'Alice' }</strong>
Query client latency: 35.8ms
Query server latency: {"parsing_ns":224023,"processing_ns":1479858,"encoding_ns":272385,"assign_timestamp_ns":1285762,"total_ns":3359570}
Number of people named "Alice": 1
</pre>